enterprise/course_catalog_api.py

```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
+"""
+Utilities to get details from the course discovery API
+"""
+from django.utils.translation import ugettext_lazy as _
+
+try:
+    from openedx.core.djangoapps.api_admin.utils import course_discovery_api_client
+    from openedx.core.djangoapps.catalog.models import CatalogIntegration
+    from openedx.core.lib.edx_api_utils import get_edx_api_data
+except ImportError:
+    course_discovery_api_client = None
+
+
+class NotConnectedToOpenEdX(Exception):
+    """
+    Exception that is raised when we try to import from
+    the core openedx repository, but it isn't available
+    """
+    pass
+
+
+def get_catalog_api_client(user):
+    """
+    Retrieve an authenticated API client that can be used to access the course
+    discovery API; raise an exception to be caught at a higher level if the
+    package doesn't have OpenEdX resources available.
+    """
+    if course_discovery_api_client is None:
+        raise NotConnectedToOpenEdX(
+            _('To get a catalog API client, this package must be installed in an OpenEdX environment.')
+        )
+    else:
+        return course_discovery_api_client(user)
+
+
+def get_all_catalogs(user):
+    """
+    Return a list of all course catalogs, including name and ID
+    """
+    client = get_catalog_api_client(user)
```

Comment:
@haikuginger suggestion: refactor these two methods into class - might be simpler to add more catalog api-related methods in future.